concrete-core/src/specification/engines/glwe_secret_key_to_relinearisation_key.rs

```diff
@@ -0,0 +1,49 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::{GlweRelinearisationKeyEntity, GlweSecretKeyEntity};
+
+engine_error! {
+    GlweSecretKeyRelinearisationConversionError for GlweSecretKeyRelinearisationConversionEngine @
+}
+
+/// A trait for engines converting a pair of GLWE secret keys into a GLWE relinearisation key
+///
+/// # Semantics
+///
+/// This [pure](super#operation-semantics) operation generates a GLWE relinearisation key
+/// containing the conversion of the two input GLWE secret keys `input1` and `input2`.
+///
+/// # Formal Definition
+///
+/// The goal of this function is to take as input two GLWE secret keys s1 and s2, and
+/// create a relinearisation key {CT_{i,j}}_{i <= j <=i, 1 <= i <=k} of the form
+///
+/// CT_{i,j} = GLev^{B,l}_s(S_i * S_j)
+pub trait GlweSecretKeyRelinearisationConversionEngine<Input1, Input2>: AbstractEngine
+where
+    Input1: GlweSecretKeyEntity,
+    Input2: GlweSecretKeyEntity,
+{
+    /// Converts a pair of GLWE secret keys to a GLWE relinearisation key
+    fn convert_glwe_secret_key_to_relinearisation_key(
+        &mut self,
+        input1: &Input1,
+        input2: &Input2,
+        // TODO: how to deal with relin-key type output here (error)/
+    ) -> Result<
+        GlweRelinearisationKeyEntity,
```

Comment:
This is a trait -> we want to return a type. Do the same thing as in `input1, input2`